system/HTTP/CURLRequest.php

```diff
@@ -555,10 +555,17 @@ protected function setCURLOptions(array $curlOptions = [], array $config = [])
                     throw HTTPException::forInvalidSSLKey($config['ssl_key']);
                 }
 
-                $curlOptions[CURLOPT_CAINFO]         = $file;
-                $curlOptions[CURLOPT_SSL_VERIFYPEER] = 1;
+                $curlOptions[CURLOPT_CAINFO] = $file;
+                if ($config['verify'] === 'yes') {
```

Comment:
There might be something I don't understand, but how can we have `yes` as a value here?

The [user guide](https://codeigniter.com/user_guide/libraries/curlrequest.html?highlight=curl#verify) says about `verify` option:

> This option describes the SSL certificate verification behavior. If the verify option is true, it enables the SSL certificate verification and uses the default CA bundle provided by the operating system. If set to false it will disable the certificate verification (this is insecure, and allows man-in-the-middle attacks!). You can set it to a string that contains the path to a CA bundle to enable verification with a custom certificate. The default value is true:

Dunno why, but we use `ssl_key` instead of `verify` when it's a string. The `ssl_key` option is not documented at all.

I feel like something has to be changed here. Either the code or the documentation.

Reply:
I missed it! Thank you.

> This option describes the SSL certificate verification behavior. If the verify option is true, it enables the SSL certificate verification and uses the default CA bundle provided by the operating system. If set to false it will disable the certificate verification (this is insecure, and allows man-in-the-middle attacks!). You can set it to a string that contains the path to a CA bundle to enable verification with a custom certificate. The default value is true:
https://codeigniter4.github.io/CodeIgniter4/libraries/curlrequest.html#verify

Reply:
Yeah... since we should be compatible with guzzle, we should use `verify`. 

When fixed, this should be noted in the changelog with some additional description, because some users could dig into the code and set `ssl_key` just to make it work.